### ISSUE TITLE
[a11y]: The contrast ratio of text with its background must meet WCAG 2.1 AA requirements

### ISSUE THREAD

**Issue by mariannebravo** [carbon-design-system/carbon#13006]
### Package

@carbon/react, carbon-components-react, @carbon/icons-react

### Browser

Chrome, Safari, Firefox

### Operating System

MacOS

### Package version

carbon-components: 10.58.3 | carbon-components-react: 7.59.4

### React version

NextJS - 12.2.4

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

 I'm using the Carbon React DatePicker using `Date.Now()` to prevent the user pick a past date (as you can see in the print below). The issue is related to the disabled options when the date is disabled, the accessibility checker is pointing a violation about text contrast, and i tried to access the CSS class using the `:global` variable from Sass, to change the disabled day color and also to try to change the background color,  and i could not solve this problem.

![image](https://user-images.githubusercontent.com/59119687/212923296-6f328709-36e1-473f-8e88-7ebb8e7cb80b.png)

### Reproduction/example

https://stackblitz.com/edit/github-fvtu7k?file=src/App.jsx

### Steps to reproduce

1. Open the example (I recommend to open in a new tab)
2. After this, open your inspect element and switch to the Accessibility Assessment tab
3. Just run the accessibility checker, and should appear this:

![image](https://user-images.githubusercontent.com/59119687/214113812-052ce8c8-e8ea-424c-b7b7-3606fbac7c48.png)

As you can see, is pointing to 22 violations errors, related to the 22 past days that are disabled.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [ ] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@mariannebravo Disabled elements are removed from the accessibility tree and are not required to meet contrast ratio requirements as seen [here in Understanding Contrast Requirements](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html).
